compositor_pipeline/src/pipeline/input/decklink/find_device.rs

```diff
@@ -2,17 +2,34 @@ use decklink::{
     get_decklinks, FlagAttributeId, IntegerAttributeId, StringAttributeId, VideoIOSupport,
 };
 
-use super::{DeckLinkError, DeckLinkOptions};
+use super::{DeckLinkError, DeckLinkInfo, DeckLinkOptions};
 
 pub(super) fn find_decklink(opts: &DeckLinkOptions) -> Result<decklink::DeckLink, DeckLinkError> {
     let decklinks = get_decklinks()?;
 
+    let decklinks_info = decklinks
+        .iter()
+        .map(|decklink| {
+            let attr = decklink.profile_attributes()?;
+            Ok(DeckLinkInfo {
+                display_name: attr.get_string(StringAttributeId::DisplayName)?,
+                persistent_id: attr
+                    .get_integer(IntegerAttributeId::PersistentID)?
+                    .map(|value| format!("{value:X}")),
+                subdevice_index: attr
+                    .get_integer(IntegerAttributeId::SubDeviceIndex)?
+                    .map(|i| i as u32),
+            })
+        })
+        .collect::<Result<_, DeckLinkError>>()?;
+
     for mut decklink in decklinks.into_iter() {
         if is_selected_decklink(opts, &mut decklink)? {
             return Ok(decklink);
         }
     }
-    Err(DeckLinkError::NoMatchingDeckLink)
+
+    Err(DeckLinkError::NoMatchingDeckLink(decklinks_info))
```

Comment:
This error message will need to include a persistent id otherwise users have no good way to find their decklink id easily.

An alternative approach is running an example from decklink crate, but this is not sth we want to recommend to users in general.